src/balancereader/src/main/java/anthos/samples/financedemo/balancereader/BalanceReaderController.java

```diff
@@ -191,28 +194,31 @@ public ResponseEntity liveness() {
      */
     @GetMapping("/balances/{accountId}")
     public ResponseEntity<?> getBalance(
-            @RequestHeader("Authorization") String bearerToken,
-            @PathVariable String accountId) {
-        LOGGER.fine("request from: " + accountId);
+        @RequestHeader("Authorization") String bearerToken,
+        @PathVariable String accountId) {
+
         if (bearerToken != null && bearerToken.startsWith("Bearer ")) {
             bearerToken = bearerToken.split("Bearer ")[1];
         }
         try {
             DecodedJWT jwt = verifier.verify(bearerToken);
             // Check that the authenticated user can access this account.
             if (!accountId.equals(jwt.getClaim("acct").asString())) {
+                LOGGER.error("Not authorized");
```

Comment:
This logging level should not be `error`

This situation is catching an incorrect login token. This is the fault of the user, and the application is actually operating exactly as it is supposed to when it raises this error.

Instead it should be either `warn` (if you believe this is unexpected behavior) or `info` (because while the application is actually operating as expected when refusing incorrect authentication, it would be nice to know that the balance could not be retrieved). ALSO, you could update the log message to be more descriptive:  e.g. "Failed to retrieve account balance:  not authorized"

Same thing ~8 lines below